aiida_kkr/workflows/BandStructure_WF.py

```diff
@@ -0,0 +1,443 @@
+        try:
+            self.report("INFO: create Bandstructure results nodes. BS calc retrieved node={}".
+                        format(self.ctx.BS_run.outputs.retrieved))
+            has_BS_run = True
+        except AttributeError as e:
+            self.report("ERROR: no Bandstructure calc retrieved node found")
+            self.report("Caught AttributeError {}".format(e))
+            has_BS_run = False
+        
+        if has_BS_run:
+            BS_retrieved = self.ctx.BS_run.outputs.retrieved
+            
+        ef = self.ctx.fermi_energy  # in eV unit
+        kpoints = self.ctx.BS_kpoints
+        outdict = {}
+        if has_BS_run:
+            ArraData = parse_BS_data(BS_retrieved, Float(ef), kpoints)
+            outdict['BS_Data'] = ArraData['BS_Data'] 
+
+        # link to the BS output nodes
+        link_nodes = outdict.copy()
+        if has_BS_run: link_nodes['BS_calc_remote'] =self.ctx.BS_run.outputs.remote_folder
+        outdict['results_wf'] = create_out_dict_node(outputnode, **link_nodes)
+
+    # create links to output nodes
+        for link_name, node in outdict.items():
+            self.out(link_name, node)
+
+        self.report("INFO: done with BS_workflow!\n")
+
+    def ev_To_Ry(value):
+        return float(value/13.6056980659)
+    
+
+@calcfunction
+def parse_BS_data(retrieved_folder, fermi_level, kpoints):
+    from masci_tools.io.common_functions import get_Ry2eV
+    from aiida.plugins import DataFactory
+    import numpy as np
+
+    eVscale = get_Ry2eV()  
+    
+    retrieved_list = retrieved_folder.list_object_names()
+    qdos_file_list = [i for i in retrieved_list if 'qdos.' in i]
+    q_vec_file = 'qvec.dat'
+
+    if q_vec_file in retrieved_list:
+        file_opened = retrieved_folder.open(q_vec_file)
+        q_vec = np.loadtxt(file_opened, skiprows=1 )
+
+    else:
+        print('{} file is not present in the retrived folder'.format(q_vec_file))
+    no_q_vec = len(q_vec[:,0])
+
+    num_dos_files = len(qdos_file_list)
+
+    with retrieved_folder.open(qdos_file_list[0]) as f:
+        total_dos = np.loadtxt(f)
+
+    for i in qdos_file_list[1:]:
+        with retrieved_folder.open(i) as f:
+            loaded_file = np.loadtxt(f)
+            total_dos[:,5:] += loaded_file[:,5:]
+    
+    ef = fermi_level.value # in eV unit
+    total_dos[:,0] = total_dos[:,0]*eVscale - ef*np.ones(np.shape(total_dos[:,0]))
+    eng_points = set(total_dos[:,0])
+    no_eng_points = len(eng_points)
+
+    dos_intensity = np.ndarray(shape = (no_eng_points, no_q_vec))
+    for ne in range(np.shape(dos_intensity)[0]):
+        nk = np.shape(dos_intensity)[1]
+
+        dos_intensity[ne,:] = total_dos[ ne*nk : (ne+1)*nk, 5 ]/eVscale 
+
+    dos_intensity = dos_intensity.T # setting eng-kpts corresponds to x-y asix
+    q_vec = np.asarray(q_vec) # converting q_vec into array
+    eng_points = np.asarray(list(eng_points)) # converting eng_popints into array
```

Comment:
since `eng_points` is a set before, you should make sure to sort the array afterwards